admin-ui/plugins/auth-server/components/AuthN/LdapForm.js

```diff
@@ -0,0 +1,209 @@
+import React, { useState, useCallback } from 'react'
+import { useFormik } from 'formik'
+import { Form, FormGroup } from 'Components'
+import GluuInputRow from 'Routes/Apps/Gluu/GluuInputRow'
+import GluuToogleRow from 'Routes/Apps/Gluu/GluuToogleRow'
+import GluuCommitFooter from 'Routes/Apps/Gluu/GluuCommitFooter'
+import GluuCommitDialog from 'Routes/Apps/Gluu/GluuCommitDialog'
+import { useDispatch } from 'react-redux'
+import { buildLdapPayload } from '../../helper'
+import { STRINGS, ACTIONS, FEATURES } from '../../helper/constants'
+
+function LdapForm({ ldapConfig: initialValues, isEdit, onSuccessApply }) {
+  const [modal, setModal] = useState(false)
+  const formik = useFormik({
+    initialValues,
+    onSubmit: (values, { setSubmitting }) => {
+      setModal(true)
+      setSubmitting(false)
+    },
+  })
+
+  const dispatch = useDispatch()
+  const handleDialogAccept = useCallback(
+    (userMessage) => {
+      const payload = buildLdapPayload(formik.values, userMessage)
+      dispatch({ type: isEdit ? ACTIONS.EDIT_LDAP : ACTIONS.ADD_LDAP, payload, onSuccessApply })
+      setModal(false)
+    },
+    [formik.values, isEdit, dispatch],
```

Comment:
Add `onSuccessApply` to the dependency array

The `handleDialogAccept` callback depends on `onSuccessApply` but it's missing from the dependency array. This could lead to stale closure issues if the prop changes.



```diff
-    [formik.values, isEdit, dispatch],
+    [formik.values, isEdit, dispatch, onSuccessApply],
```